src/NUnitFramework/tests/Constraints/NumericsTest.cs

```diff
@@ -83,6 +84,21 @@ public void CanMatchDecimalWithPercentage()
             Assert.IsTrue(Numerics.AreEqual(10000m, 10500m, ref tenPercent));
         }
 
+        [Test]
+        public void CanCalculateAbsoluteDifference()
+        {
+            Assert.AreEqual(Numerics.Difference(10000m, 9500m, absoluteTolerance), 500m);
+            Assert.AreEqual(Convert.ToDouble(Numerics.Difference(0.1, 0.05, absoluteTolerance)), 0.05, 0.00001);
+            Assert.AreEqual(Convert.ToDouble(Numerics.Difference(0.1, 0.15, absoluteTolerance)), -0.05, 0.00001);
```

Comment:
I think the "expected" and the "actual" arguments have been swapped in this test (and in the test below). The signature of `Assert.AreEqual` is `Assert.AreEqual(double expected, double actual, double tolerance);`. On could also use the constraint based model instead, as this avoids the confusion of `Assert.AreEqual`. E.g. the first assertion could be written as `Assert.That(Numerics.Difference(10000m, 9500m, absoluteTolerance), Is.EqualTo(500m))`.

Also I reviewed this file first in the PR and was a bit confused how the tolerance would affect the the difference, e.g. what is the difference between `10000m` and `9500m` when the tolerance is `0.1` ? (I now know that we only use the mode of the tolerance, so I think it would make more sense just to pass in `ToleranceMode` and not `Tolerance` to `Difference`.

Reply:
Agree! Refactored.